tensorboard/tools/diagnose_me.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Self-diagnosis script. Run this if your environment is broken."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# This script may only depend on the Python standard library. It is not
+# built with Bazel and should not assume any third-party dependencies.
+import collections
+import errno
+import functools
+import hashlib
+import inspect
+import logging
+import os
+import pipes
+import shlex
+import socket
+import subprocess
+import sys
+import tempfile
+import textwrap
+import traceback
+
+
+# A *check* is a function (of no arguments) that performs a diagnostic,
+# writes log messages, and optionally yields suggestions. Each check
+# runs in isolation; exceptions will be caught and reported.
+CHECKS = []
+
+
+# A suggestion to the end user.
+#   headline (str): A short description, like "Turn it off and on
+#     again". Should be imperative with no trailing punctuation. May
+#     contain inline Markdown.
+#   description (str): A full enumeration of the steps that the user
+#     should take to accept the suggestion. Within this string, prose
+#     should be formatted with `reflow`. May contain Markdown.
+Suggestion = collections.namedtuple("Suggestion", ("headline", "description"))
+
+
+def check(fn):
+  """Register a function as a check.
```

Comment:
Can you add a line that this is intended to be used as a decorator, i.e., as @check.

Reply:
Sure.